tests/bank_interface/test_bank_client.py

```diff
@@ -0,0 +1,92 @@
+from typing import cast
+from unittest.mock import Mock, call
+
+from personal_finances.bank_interface.bank_client import GocardlessClient
+from personal_finances.bank_interface.nordigen_adapter import NordigenTransactions
+
+
+def get_nordigen_mock() -> Mock:
+    nordigen_mock = Mock()
+
+    nordigen_mock.requisition.get_requisition_by_id.__name__ = "get_requisition_by_id"
+    nordigen_mock.requisition.get_requisition_by_id.side_effect = [
+        {"accounts": ["account11", "account12"]},
+        {"accounts": ["account21", "account22"]},
+    ]
+
+    nordigen_mock.account_api.__name__ = "account_api"
+    nordigen_mock.account_api.return_value.get_transactions.__name__ = (
+        "get_transactions"
+    )
+    nordigen_mock.account_api.return_value.get_transactions.side_effect = [
+        {
+            "transactions": {
+                "booked": ["transaction_b11", "transaction_b12"],
+                "pending": ["transaction_p11"],
+            }
+        },
+        {
+            "transactions": {
+                "booked": ["transaction_b21"],
+                "pending": ["transaction_p21", "transaction_p22"],
+            }
+        },
+        {
+            "transactions": {
+                "booked": ["transaction_b31"],
+                "pending": ["transaction_p31"],
+            }
+        },
+        {
+            "transactions": {
+                "booked": ["transaction_b41", "transaction_b42"],
+                "pending": ["transaction_p41"],
```

Comment:
What about testing empty lists for "booked" and "pending". I know its gonna be difficult to happen a user has no booked transaction, but for pending may can happen

Reply:
Good one, I'll add it in a next revision